### ISSUE TITLE
Script is killed after a while

### ISSUE THREAD

**Issue by rsc9421** [danielhavir/voxelize3d#2]
I tried to use it but doesn't work, the script cannot run the function and is killed after a while, both voxelize and voxelize_jit

**Comment by Chinese-Coding** [comment]
Yes, really? I also want to write some code to replace these code in spconv as this package can't adapt to NumPy 2.0. I currently have my own dataset that I can use to determine if its code is correct, as well as to do some performance testing. This work shouldn't be very complicated.

**Comment by danielhavir** [comment]
Tested this on [NuScenes](https://www.nuscenes.org/) pointcloud with with NumPy version `2.1.3` and Numba version `0.61.0` with no issues:

```
import numpy as np
from voxelize import voxelize
from jit_voxelize import voxelize_jit

path = ...  # path to a .pcd.bin
points = np.fromfile(path, dtype=np.float32).reshape(-1, 5)[:, :4]
voxels, coords, num_points_per_voxel = voxelize(points, voxel_size=np.array([0.2, 0.2, 0.4]), grid_range=np.array([-50, -50, -5, 50, 50, 3]), max_points_in_voxel=10, max_num_voxels=50000)

voxels, coords, num_points_per_voxel = voxelize_jit(points, voxel_size=np.array([0.2, 0.2, 0.4]), grid_range=np.array([-50, -50, -5, 50, 50, 3]), max_points_in_voxel=10, max_num_voxels=50000)
```

JIT version circa 10.8x faster on MacBook M1 Pro

If you provide a repro for the issue, I'm happy to inspect.

**Comment by Chinese-Coding** [comment]
Enabling the JIT feature does indeed improve performance, and I’ve tested this part. However, compared to using the SpConv library, it’s still quite a bit slower.
As I tested, using the SpConv package to process about 58,000 points takes 0.00587 seconds, while using the NumPy JIT it takes 0.0598 seconds. Obviously, SpConv is still much faster, and this feels really frustrating.

**Comment by danielhavir** [comment]
Got it. When I wrote this 5 years ago, the performance was on par, but not surprised that SpConv outperforms nowadays. The reason I wrote this was to remove the SpConv dependency at the time.

Thank you for testing it, I updated the README.

**Comment by Chinese-Coding** [comment]
I conducted additional tests and discovered an error in my approach. In the previous test, I performed only a single trial and used the warm-up time as the result for NumPy. I have since written a loop to calculate the mean time consumption. (Before entering the loop, I warmed up the NumPy code.) The results indicate that the time consumed by the NumPy code with JIT is comparable to that of using spconv.
spconv mean time: 0.002035832643508911 seconds
numpy mean time: 0.005135119915008545 seconds
warmed up time: 0.2559943199157715 seconds

**Comment by danielhavir** [comment]
Thanks for testing these. Seems reasonable. I have a NumPy version that's about 30% faster (circa 3.5ms mean) but the output is not identical to SpConv as it changes the order in which points are assigned to voxels and can—depending on parameters—return different set of voxels.
